mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/IMantisPersistenceProvider.java

```diff
@@ -21,18 +21,25 @@
 import io.mantisrx.master.jobcluster.job.IMantisStageMetadata;
 import io.mantisrx.master.jobcluster.job.worker.IMantisWorkerMetadata;
 import io.mantisrx.master.resourcecluster.DisableTaskExecutorsRequest;
+import io.mantisrx.server.core.domain.JobArtifact;
 import io.mantisrx.server.master.domain.JobClusterDefinitionImpl.CompletedJob;
 import io.mantisrx.server.master.persistence.exceptions.InvalidJobException;
 import io.mantisrx.server.master.resourcecluster.ClusterID;
 import io.mantisrx.server.master.resourcecluster.TaskExecutorID;
 import io.mantisrx.server.master.resourcecluster.TaskExecutorRegistration;
 import java.io.IOException;
+import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 import rx.Observable;
 
 
-public interface IMantisStorageProvider {
+/**
+ * A way to persist mantis master related metadata to a durable storage.
+ * See {@link KeyValueBasedPersistenceProvider} for how mantis job cluster,
+ * mantis job info is persisted to a key-value based storage (like cassandra)
+ */
+public interface IMantisPersistenceProvider {
```

Comment:
FYI @sundargates has been pushing for moving all mantis interfaces not to have the "I" prefix.

Reply:
I'd like to take care of it in a separate PR.